elastic/kibana
# [Actionable Observability] Integrate alert search bar on rule details page

Resolves #143962

## 📝 Summary
In this PR, an alerts search bar was added to the rule details page by syncing its state to the URL. This will enable navigating to the alerts table for a specific rule with a filtered state based on active or recovered.
### Notes
- Renamed alert page container to alert search bar container and used it both in alerts and rule details page (it will be responsible to sync search bar params to the URL) --> moved to a shared component
- Moved AlertsStatusFilter to be a sub-component of the shared observability search bar
- Allowed ObservabilityAlertSearchBar to be used both as a stand-alone component and as a wired component with syncing params to the URL (ObservabilityAlertSearchBar, ObservabilityAlertSearchbarWithUrlSync)
- Set a minHeight for the Alerts and Execution tab, otherwise, the page will have extra scroll on the tab change while content is loading (very annoying!)

## 🎨 Preview
![image](https://user-images.githubusercontent.com/12370520/200547324-d9c4ef3c-8a82-4c16-88bd-f1d4b2bc8006.png)

## 🧪 How to test
- Create a rule and go to the rule details page
- Click on the alerts tab and change the search criteria, you should be able to see the criteria in the query parameter
- Refresh the page, alerts tab should be selected and you should be able to see the filters that you applied in the previous step
- As a side test, check alert search bar on alerts page as well, it should work as before
